charts/tekton-apps/templates/_snippets.tpl

```diff
@@ -215,3 +214,10 @@ the way you use it:
   value: {{ .default }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Check what namespace need to use, from project or component or extraApp field
+*/}}
+{{- define "tekton-apps.set-namespace-from-component-or-project" -}}
+{{- or ((.component).argocd).destinationNamespace ((.project).argocd).namespace | required "Error: One of the following should be set: .project.argocd.namespace, .component.argocd.destinationNamespace, or .extraApp.argocd.destinationNamespace" }}
```

Comment:
Don't think that we need to add info about `extraApp` here, basically current function knows nothing about `extraApp` at all and it is strange to see it here (it is specific to only one function call from `argocd_extra_applications.yaml` file). And snippet should know nothing about that call (that it passed `extraApp` content as component).

So let's add one more parameter for current snippet -  `.component_name`, so in the include you will pass also "component_name" "component" or "component_name" "extraApp" (depending on what dict you pass to `component` variable). And here in `required` message you will dynamically add either `.component.argocd.destinationNamespace` or `.extraApp.argocd.destionationNamespace` (use `printf "%s-%s"` for that).

Reply:
Thanks for helping! I rewrote error message